build.gradle

```diff
@@ -38,14 +38,10 @@ subprojects {
     configurations.all {
         resolutionStrategy {
             failOnVersionConflict()
-            // Force the IDEA SDK provided version of these dependencies to avoid classloading issues.
-            force 'org.apache.httpcomponents:httpclient:4.4.1', 'com.google.guava:guava:17.0'
```

Comment:
I'm actually not worried about the dep count (yet).  My concern is that the
'force.. ' statements I put in place were to fix bugs in class version
conflict between local versions of the libraries and IJs.  Removing them
allows developers to inadvertently reintroduce these bugs that are tricky
and only visible at runtime.  So I think we should do one of two things:
1) completely disallow the inclusion of any version of a dependency
provided by IJ
2) force any dependency that gets duplicated to be the same version as IJ

option 1 seems like the better solution since we don't run the risk of
going out of sync with the actual versions provided by the IJ SDK.

On Tue, Jan 12, 2016 at 4:44 PM, Appu Goundan <notifications@github.com>
wrote:

> In build.gradle
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/349#discussion_r49518444>
> :
>
> > @@ -38,14 +38,10 @@ subprojects {
> >      configurations.all {
> >          resolutionStrategy {
> >              failOnVersionConflict()
> > -            // Force the IDEA SDK provided version of these dependencies to avoid classloading issues.
> > -            force 'org.apache.httpcomponents:httpclient:4.4.1', 'com.google.guava:guava:17.0'
>
> I worry we will depend on it too much to make sure we don't add
> unnecessary additions. Perhaps a dependency count check is better? if total
> dependencies != X, then are you sure you need this dependency?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/gcloud-intellij/pull/349/files#r49518444>
> .
>